timesketch/api/v1/resources/llm.py

```diff
@@ -0,0 +1,239 @@
+        except KeyError:
+            abort(
+                definitions.HTTP_STATUS_CODE_BAD_REQUEST,
+                f"Invalid LLM feature: {feature_name}",
+            )
+
+    def _validate_indices(self, sketch: Sketch, query_filter: dict) -> list:
+        """Extracts and validates timeline IDs from the query filter for a sketch."""
+        all_indices = list({t.searchindex.index_name for t in sketch.timelines})
+        indices = query_filter.get("indices", all_indices)
+        if "_all" in indices:
+            indices = all_indices
+        indices, timeline_ids = utils.get_validated_indices(indices, sketch)
+        if not indices:
+            abort(
+                definitions.HTTP_STATUS_CODE_BAD_REQUEST,
+                "No valid search indices were found.",
+            )
+        return timeline_ids
+
+    def _generate_prompt(
+        self,
+        feature: feature_manager.LLMFeatureInterface,
+        sketch: Sketch,
+        form: dict,
+        timeline_ids: list,
+    ) -> str:
+        """Generates the LLM prompt based on the feature and request data."""
+        try:
+            return feature.generate_prompt(
+                sketch, form=form, datastore=self.datastore, timeline_ids=timeline_ids
+            )
+        except ValueError as e:
+            abort(definitions.HTTP_STATUS_CODE_BAD_REQUEST, str(e))
+
+    def _execute_llm_call(
+        self, feature: feature_manager.LLMFeatureInterface, prompt: str, sketch_id: int
+    ) -> dict:
+        """Executes the LLM call with a timeout using multiprocessing."""
+        with multiprocessing.Manager() as manager:
+            shared_response = manager.dict()
+            process = multiprocessing.Process(
+                target=self._get_content_with_timeout,
+                args=(feature, prompt, shared_response),
+            )
+            process.start()
+            process.join(timeout=self._LLM_TIMEOUT_WAIT_SECONDS)
+
+            if process.is_alive():
+                logger.warning(
+                    "LLM call timed out after %d seconds.",
+                    self._LLM_TIMEOUT_WAIT_SECONDS,
+                )
+                process.terminate()
+                process.join()
+                self.METRICS["llm_errors_total"].labels(
+                    sketch_id=str(sketch_id), feature=feature.NAME, error_type="timeout"
+                ).inc()
+                abort(definitions.HTTP_STATUS_CODE_BAD_REQUEST, "LLM call timed out.")
+
+            response = dict(shared_response)
+            if "error" in response:
+                self.METRICS["llm_errors_total"].labels(
+                    sketch_id=str(sketch_id),
+                    feature=feature.NAME,
+                    error_type="llm_api_error",
+                ).inc()
+                abort(
+                    definitions.HTTP_STATUS_CODE_INTERNAL_SERVER_ERROR,
+                    "Error during LLM processing.",
+                )
```

Comment:
Is there any actionable information we could give the user with this error message?

Reply:
Provided details of the error in the error message